File: hanoi.vb
Imports System

Public Class Hanoi
	Public Sub hanoi(n As Integer, x As String, y As String, z As String)
		If n>1 Then Call hanoi(n-1, x, z, y)
		Console.WriteLine(x & "=>" & y)
		If n>1 Then Call hanoi(n-1, z, y, x)
	End Sub
	Public Shared Sub Main()
		Dim n As Integer
		n=Console.ReadLine()
		Dim h As New Hanoi
		h.hanoi(n,"a","b","c")
	End Sub
End Class

File: hanoi.ml
let rec hanoi n x y z =
  if n > 1 then hanoi (n-1) x z y;
  print_endline ( x ^ "=>" ^ y );
  if n > 1 then hanoi (n-1) z y x;;

let n = read_line () in
hanoi (int_of_string n) "a" "b" "c";;